src/hybrid_reader.c

```diff
@@ -424,7 +424,8 @@ IndexIterator *NewHybridVectorIterator(HybridIteratorParams hParams) {
   hi->timeoutCtx = (TimeoutCtx){ .timeout = hParams.timeout, .counter = 0 };
   hi->runtimeParams.timeoutCtx = &hi->timeoutCtx;
 
-  if (hParams.childIt == NULL) {
+  if (hParams.childIt == NULL || hParams.query.k == 0) {
+    // If there is no child iterator, or the query is going to return 0 results, we can use simple KNN.
```

Comment:
If we know the query will return 0 results, is it not possible to optimize even further and short-circuit?

Reply:
only a bit more, but on 0 results it does not mean the entire query will return 0, and stuff still needs to be prepared. on the first read, we will return EOF